include/oneapi/tbb/concurrent_queue.h

```diff
@@ -135,37 +136,42 @@ class concurrent_queue {
     }
   
     concurrent_queue& operator=( const concurrent_queue& other ){
-      if (my_queue_representation != other.my_queue_representation) {
-	my_queue_representation = other.my_queue_representation;
-      }
-      return *this;
+        //TODO: implement support for std::allocator_traits::propogate_on_container_copy_assignment
+        if (my_queue_representation != other.my_queue_representation) {
+            clear();
+	    my_queue_representation->assign(*other.my_queue_representation, my_allocator, copy_construct_item);
```

Comment:
micro_queue's assign reallocates the memory and then copy and move constructor items are constructed and placed in this preallocated memory using new. I tried the test that you had mentioned (I believe you meant test_scoped_allocator) but that test fail because '"Allocator should propagate to the data during copy construction'. Do you think it is something that will get resolved when we implement support of propagate_on_container or is it something that you mentioned earlier.

Reply:
I meant not exactly the scoped allocator test. It actually tests only the support for `propagate_on_container_*`. I meant a test that checks the following:
Suppose that we have two queues with unequal allocator instances (`q1.get_allocator() != q2.get_allocator()`). 
I guess counting allocators would be good for this test (we have the implementation somewhere in tests).
First of all - save the number of allocations/deallocations done for creating the container.
Then do `q1 = q2` and check that `q1` destroys all of the elements and deallocates using the previous allocator from `q1` and then re-allocates the memory using the allocator from `q2`.
If the description seems overcomplicated, we can discuss it offline:)